backend/tests/e2e/e2e_utils.py

```
@@ -8,12 +8,78 @@
+
+        # Normalize content for comparison
+        expected_content = E2ETestUtils.normalize_content(expected_msg.get("content", ""))
+        actual_content = E2ETestUtils.normalize_content(actual_msg.get("content", ""))
+        
+        if expected_content != actual_content:
+            # Show difference for debugging
+            print(f"\n❌ Content mismatch in message {i}:")
+            print(f"  Expected length: {len(expected_content)}")
+            print(f"  Actual length: {len(actual_content)}")
+            # Find where they differ
+            for idx, (e_char, a_char) in enumerate(zip(expected_content, actual_content)):
```

Ruff (0.14.8)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)